packages/eslint-plugin/index.d.ts

```diff
@@ -0,0 +1,4 @@
+import { Linter, Rule } from 'eslint';
+
+export const rules: Record<string, Rule.RuleModule>;
+export const configs: Record<string, Linter.Config>;
```

Comment:
as mentioned in https://github.com/typescript-eslint/typescript-eslint/issues/3129#issuecomment-788269448 - we do not have a dependency on the eslint types - purposely so.
We also recommend consumers *don't* use those types if you're writing typescript rules because they will be wrong incorrect. 

We declare our own types for this project which properly declare the expected options and nodes.

```suggestion
import { TSESLint } from '@typescript-eslint/experimental-utils';

export const rules: Record<string, TSESLint.Rule.RuleModule>;
export const configs: Record<string, TSESLint.Linter.Config>;
```